### PR TITLE
3942 dcm notifications

### PR DESCRIPTION
**Pull request by pdurbin**
## Related Issues

- connects to #3942 DCM -> DV user notification APIs

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: DatasetsIT, DataCaptureModuleServiceBeanIT
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: https://github.com/IQSS/dataverse/tree/develop/scripts/database/upgrades
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/4.6.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

**Comment by pdurbin** [comment]
Here's the todo list/punch list from the meeting we just had:

img_20170801_154106


Also, @pameyer supplied existing messages:

happy path:
Dataset doi:10.15785/SBGRID/480has been successfully uploaded and verified.  If you have any questions, please contact data@sbgrid.org.

failure path:
One or more files in your dataset failed to match the checksum calculated on your local system.  Please re-run the upload script associated with this dataset.  If you have any questions, please contact data@sbgrid.org.

File: src/main/java/edu/harvard/iq/dataverse/api/Datasets.java [hunk 1]
```diff
@@ -613,6 +627,72 @@ public Response getRsync(@PathParam("identifier") String id) {
             return error(Response.Status.INTERNAL_SERVER_ERROR, "Something went wrong attempting to download rsync script: " + EjbUtil.ejbExceptionToString(ex));
         }
     }
+    
+    @POST
+    @Path("{identifier}/dataCaptureModule/checksumValidation")
+    public Response receiveChecksumValidationResults(@PathParam("identifier") String id, JsonObject jsonFromDcm) throws IOException {
```
Comment (oscardssmith) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: where does this throw?

Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: Whoops. I'll take that out as well as some imports I added when adding test code. Good catch, @oscardssmith . Thanks!

Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: Fixed in f3048e0.

File: src/main/java/edu/harvard/iq/dataverse/api/Datasets.java [hunk 2]
```diff
@@ -613,6 +622,72 @@ public Response getRsync(@PathParam("identifier") String id) {
...
+        if (!authenticatedUser.isSuperuser()) {
+            return error(Response.Status.FORBIDDEN, "Superusers only.");
+        }
+        String statusMessageFromDcm = jsonFromDcm.getString("status");
+        try {
+            Dataset dataset = findDatasetOrDie(id);
+            if ("validation passed".equals(statusMessageFromDcm)) {
+                try {
+//                    String url = "http://localhost:8080/api/batch/jobs/import/datasets/files/${DOI_SHOULDER}/${datasetIdentifier}?mode=MERGE&uploadFolder=trn&totalSize=${sz}&userId=${dv_userId}";
+                    String url = systemConfig.getDataverseSiteUrl() + "/api/batch/jobs/import/datasets/files/" + dataset.getId();
```
Comment (pameyer) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: Why not leave this as calling localhost?

Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/api/Datasets.java]: @pameyer no objection. Fixed in 6c9c249.

**Pull request merged**